### PR TITLE
Upmerge 1.14.x into master

### PR DESCRIPTION
**Pull request by mehrdada**

### CODE REVIEW

**Comment by mehrdada** [comment]
@jtattermusch please take a look at `src/csharp/build_packages_dotnetcli.sh` (esp the "1.14.0" part. why is that not regenerated with `regenerate_projects.sh`?)

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,980,675      Total (=)      1,980,675

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,811,768      Total (>)     10,811,761

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: src/csharp/build_packages_dotnetcli.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
```
Comment (jtattermusch) [inline src/csharp/build_packages_dotnetcli.sh]: this file has been deleted in upstream/master. Let's delete from this PR as well.

Comment (jtattermusch) [reply src/csharp/build_packages_dotnetcli.sh]: FTR https://github.com/grpc/grpc/pull/16208/files#diff-f8017b26a71fa84b0b616110830f9858

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,981,730      Total (=)      1,981,730

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,814,018      Total (>)     10,814,015

 No significant differences in binary sizes


```

**Pull request merged**